lib/tapioca/generic_type_registry.rb

```diff
@@ -150,6 +150,32 @@ def register_type_variable(constant, type_variable_type, type_variable, fixed, l
         )
       end
 
+      sig { params(constant: Class).returns(Class) }
+      def create_sealed_safe_subclass(constant)
+        # If the constant is not sealed let's just bail early.
+        # We just return a subclass of the constant.
+        return Class.new(constant) unless T::Private::Sealed.sealed_module?(constant)
+
+        # Since sealed classes can normally not be subclassed, we need to trick
+        # sealed classes into thinking that the generic type we are
+        # creating by subclassing is actually safe for sealed types.
+        #
+        # Get the filename the sealed class was declared in
+        decl_file = constant.instance_variable_get(:@sorbet_sealed_module_decl_file)
+        begin
```

Comment:
Why do we need the `begin`/`ensure` blocks here? Are we expecting something to fail?

Reply:
The `ensure` is to set the state of `@sorbet_sealed_module_decl_file` back to its original value after we've changed it in the `begin` block.

Reply:
Since calling `sealed!` updates that value and before that we've removed it as well.

Reply:
If nothing is raising from what's inside the `begin`, do we need the `ensure`? Could this just be a plain sequence of instructions instead of a begin block?

Reply:
I just want to be sure that if anything *does* go wrong, the previous state will always be restored. Imagine that the `sealed!` call inside the block throws an error for some reason.